tests/sqlparser_common.rs

```diff
@@ -5976,15 +5976,6 @@ fn lateral_derived() {
     chk(false);
     chk(true);
 
-    let sql = "SELECT * FROM customer LEFT JOIN LATERAL generate_series(1, customer.id)";
```

Comment:
Can you please update this test to show this is now parsing correctly?

And maybe add a test with some other syntax that doesn't work?

Like
```sql
SELECT * FROM LATERAL UNNEST ([10,20,30]) as numbers WITH OFFSET;
```

Reply:
Added